### PR TITLE
fix: A bug in the stream page causing 4th biggest 5xx error on Next in the past 24h

### PR DESCRIPTION
**Pull request by matthew-andrews**
`.reduce` needs an `initialValue`

### CODE REVIEW

File: server/controllers/related/tag.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = function (req, res) {
 				)
 				// flatten
 				.reduce(
-					(prev, relations) => prev.concat(relations)
+					(prev, rels) => prev.concat(rels), []
```
Comment (matthew-andrews) [inline server/controllers/related/tag.js]: renamed this to make it clear it's a different array

File: server/controllers/related/tag.js [hunk 2]
```diff
@@ -47,7 +47,8 @@ module.exports = function (req, res) {
 				relations: relations
 			});
 		})
-		.catch(
-			() => res.status(500).send()
-		);
+		.catch(err => {
+			logger.error({ event: 'RELATED_SPECIAL_REPORTS_ERROR', err });
```
Comment (matthew-andrews) [inline server/controllers/related/tag.js]: added a logger so that we can see this in the logs

**Comment by bjfletcher** [comment]
```
> let t = [1, 2, 3].map(t => t+1);
undefined
> t
[2, 3, 4]
```

madness

this is a great example of why immutability/FP school of thought helps us a lot around hard-to-find bugs

we should also perhaps do an audit of all our codebase, replacing `let` where it is meant to be `const`

**Comment by matthew-andrews** [comment]
do you agree with my changes @bjfletcher ?

**Comment by matthew-andrews** [comment]
your code snippet makes sense to me ……

**Comment by bjfletcher** [comment]
@matthew-andrews yes, most absolutely... apart from the logging change suggested by Darren

File: server/controllers/related/tag.js (hunk 2 above)
Comment (matthew-andrews) [inline server/controllers/related/tag.js]: oo

**Pull request merged**